### ISSUE TITLE
[Accessibility]: Chart "show as table" button is not keyboard accessible

### ISSUE THREAD

**Issue by amortiz** [carbon-design-system/carbon-charts#2037]
### Application/Team

IBM Cloud infrastructure

### Environment

Browser

### What happened? What did you expect to happen instead?

If you tab to the "Show as table" button on any chart, and press enter/space, the modal does not open (it only opens if you click it with a mouse click, so it is not keyboard accessible)

### What WCAG 2.1 checkpoint does the issue violate?

unsure

### Version

Latest, just try any chart

### Data & options used

```js

```

### Relevant log output

```js

```

### What priority level would this issue be in your opinion?

P3
